allennlp/models/biattentive_classification_network.py

```diff
@@ -162,16 +164,14 @@ def forward(self,  # type: ignore
         mean_pool = torch.sum(integrated_encodings, 1) / torch.sum(text_mask, 1, keepdim=True)
 
         # Self-attentive pooling layer
-        flat_self_attentive_logits = self._self_attentive_pooling_projection(
-                integrated_encodings.contiguous().view(-1, self._integrator.get_output_dim()))
-        self_attentive_logits = flat_self_attentive_logits.view(batch_size, sequence_length)
+        # Run through linear projection. Shape: (batch_size, sequence length, 1)
+        # Then remove the last dimension to get the proper attention shape (batch_size, sequence length).
+        self_attentive_logits = self._self_attentive_pooling_projection(
+                integrated_encodings.contiguous()).squeeze(2)
```

Comment:
Ah, great, this makes much more sense.  It sure looks like you shouldn't need this `.contiguous()`, though.